README.md

```diff
@@ -1,12 +1,28 @@
-# Zeto, a UTXO based privacy-preserving token toolkit using Zero Knowledge Proofs
+# Zeto - UTXO based privacy-preserving token toolkit using Zero Knowledge Proofs
 
 This project hosts the multiple patterns to implement privacy preserving tokens on EVM. The patterns all share the same basic architectural concepts:
 
 - **Transaction model**: the UTXO model is adopted instead of the account model, for better support of parallel processing. Due to the necessity of maintaining private states offchain in order to achieve privacy, the client must continuously keep their private states in sync with the smart contract. Using an account model makes it more difficult to achieve this because incoming transfers from other parties would invalidate an account's state, making the account owner unable to spend from its account unless the private state has been sync'ed again. Solutions to this issue, often referred to as front-running, typically involve a spending window with a pending queue, which result in limited parallel processing of transactions from the same spending account. With a UTXO model, each state is independent of the others, so parallel processing is better achieved.
 - **Commitments**: each UTXO is tracked by the smart contract as a hash, or commitment, of the following components: value, salt and owner public key
 - **Finality**: each transaction's validity is verified by the smart contract before allowing the proposed input UTXOs to be nullified and the output UTXOs to come into existence. In other words, this is not an optimistic design and as such does not rely on a multi-day challenge period before a transaction is finalized. Every transaction is immediately finalized once it's mined into a block.
 
-# Enforcing token transfer policies with zero knowledge proofs
+# Overview of how Zeto tokens work
+
+The following diagram illustrates the basics of Zeto tokens.
+
+![Zeto token basics](/resources/c-utxo-zkp-1.jpg)
+
+- Party A owns 3 Zeto tokens at the beginning: `#1, #2, #3`. The 3 tokens have been minted in the Zeto smart contract and represented by their commitments, or `hash(value, owner public key, salt)`
+  - As the owner of the tokens, party A also has access to the secrets that the commitments can be opened to, namely the value and salt. The secrets are represented as private states: `s1, s2, s3`
+  - How party A obtained the secrets for the owned tokens, is dependent on the specific Zeto implementation. Tt can be from offchain channels or from onchain (encrypted) data
```

Comment:
```suggestion
  - How party A obtained the secrets for the owned tokens, is dependent on the specific Zeto implementation. It can be from offchain channels or from onchain (encrypted) data
```